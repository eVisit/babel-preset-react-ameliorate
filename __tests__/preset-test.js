const babel = require("@babel/core"),
      FS = require('fs'),
      PATH = require('path'),
      presetReactAmeliorate = require('../src');

// Date mock
require('jest-mock-now')();

const baseProps = {
  id: '__ID__',
  customPropString: 'hello world!',
  customPropBoolean: true,
  customPropNumber: 5,
  customPropArray: [
    5,
    true,
    "derp!!!"
  ],
  customPropObject: {
    sweet: 'stuff',
    awesome: 453,
    dude: 'crazy!'
  }
};

function transform(code, extraOpts, extraBabelOpts) {
  return new Promise((resolve, reject) => {
    var babelOpts = Object.assign({
      code: true,
      ast: false,
      babelrc: false,
      presets: [
        [presetReactAmeliorate, Object.assign({}, extraOpts || {})]
      ]
    }, extraBabelOpts);

    babel.transform(
      ('' + code),
      babelOpts,
      function(error, result) {
        if (error)
          return reject(error);

        resolve(result);
      }
    );
  });
}

function transformFile(fileName, extraOpts) {
  var fullFileName = (PATH.join(__dirname, 'test-code', fileName) + '.js');
  return transform(FS.readFileSync(fullFileName), extraOpts, { filename: fullFileName });
}

describe('babel-preset-react-ameliorate', function () {
  it('should work with default options', async function() {
    var result = await transformFile('test01');
    expect(result.code).toMatchSnapshot();
  });

  it('should work with plugin options', async function() {
    var result = await transformFile('test01', {
      pluginOptions: {
        'babel-plugin-transform-react-ameliorate-jsx': {
          pragma: '__somethingElseBro(this)'
        }
      }
    });

    expect(result.code.indexOf('__somethingElseBro')).toBeGreaterThan(-1);
    expect(result.code).toMatchSnapshot();
  });

  it('should work allow adding extra JSX properties', async function() {
    var result = await transformFile('test01', {
      pluginOptions: {
        'transform-react-jsx-properties': {
          elements: [
            {
              test: '^.*$',
              props: baseProps
            },
            {
              test: '^div$',
              props: { elementType: 'div' }
            },
            {
              test: '^span$',
              props: { elementType: 'span' }
            }
          ]
        }
      }
    });

    expect(result.code.indexOf('customPropString')).toBeGreaterThan(-1);
    expect(result.code.indexOf('customPropBoolean')).toBeGreaterThan(-1);
    expect(result.code.indexOf('customPropNumber')).toBeGreaterThan(-1);
    expect(result.code.indexOf('customPropArray')).toBeGreaterThan(-1);
    expect(result.code.indexOf('derp!!!')).toBeGreaterThan(-1);
    expect(result.code.indexOf('sweet')).toBeGreaterThan(-1);
    expect(result.code.indexOf('stuff')).toBeGreaterThan(-1);
    expect(result.code.indexOf('awesome')).toBeGreaterThan(-1);
    expect(result.code.indexOf('453')).toBeGreaterThan(-1);
    expect(result.code.indexOf('dude')).toBeGreaterThan(-1);
    expect(result.code.indexOf('crazy!')).toBeGreaterThan(-1);
    expect(result.code).toMatchSnapshot();
  });
});
