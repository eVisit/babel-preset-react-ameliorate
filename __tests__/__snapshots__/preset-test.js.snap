// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-preset-react-ameliorate should work allow adding extra JSX properties 1`] = `"var _interopRequireDefault=require(\\"@babel/runtime/helpers/interopRequireDefault\\");var _extends2=_interopRequireDefault(require(\\"@babel/runtime/helpers/extends\\"));var _classCallCheck2=_interopRequireDefault(require(\\"@babel/runtime/helpers/classCallCheck\\"));var _createClass2=_interopRequireDefault(require(\\"@babel/runtime/helpers/createClass\\"));var _possibleConstructorReturn2=_interopRequireDefault(require(\\"@babel/runtime/helpers/possibleConstructorReturn\\"));var _getPrototypeOf2=_interopRequireDefault(require(\\"@babel/runtime/helpers/getPrototypeOf\\"));var _inherits2=_interopRequireDefault(require(\\"@babel/runtime/helpers/inherits\\"));var _react=_interopRequireDefault(require(\\"react\\"));var MyTestComponent=function(_React$Component){(0,_inherits2.default)(MyTestComponent,_React$Component);function MyTestComponent(){(0,_classCallCheck2.default)(this,MyTestComponent);return(0,_possibleConstructorReturn2.default)(this,(0,_getPrototypeOf2.default)(MyTestComponent).apply(this,arguments));}(0,_createClass2.default)(MyTestComponent,[{key:\\"render\\",value:function render(){var opts={test:true,derp:'something',func:function func(){}};return _react.default.createElement(\\"div\\",(0,_extends2.default)({hello:true},opts,{id:\\"1479427200000\\",customPropString:\\"hello world!\\",customPropBoolean:true,customPropNumber:5,customPropArray:[5,true,\\"derp!!!\\"],customPropObject:{sweet:\\"stuff\\",awesome:453,dude:\\"crazy!\\"},elementType:\\"div\\"}),_react.default.createElement(_react.default.Fragment,null,_react.default.createElement(\\"span\\",{stuff:true,id:\\"1479427200001\\",customPropString:\\"hello world!\\",customPropBoolean:true,customPropNumber:5,customPropArray:[5,true,\\"derp!!!\\"],customPropObject:{sweet:\\"stuff\\",awesome:453,dude:\\"crazy!\\"},elementType:\\"span\\"},\\"Hello world!\\"),_react.default.createElement(\\"derp\\",{id:\\"1479427200002\\",customPropString:\\"hello world!\\",customPropBoolean:true,customPropNumber:5,customPropArray:[5,true,\\"derp!!!\\"],customPropObject:{sweet:\\"stuff\\",awesome:453,dude:\\"crazy!\\"}})));}}]);return MyTestComponent;}(_react.default.Component);"`;

exports[`babel-preset-react-ameliorate should work with default options 1`] = `"var _interopRequireDefault=require(\\"@babel/runtime/helpers/interopRequireDefault\\");var _extends2=_interopRequireDefault(require(\\"@babel/runtime/helpers/extends\\"));var _classCallCheck2=_interopRequireDefault(require(\\"@babel/runtime/helpers/classCallCheck\\"));var _createClass2=_interopRequireDefault(require(\\"@babel/runtime/helpers/createClass\\"));var _possibleConstructorReturn2=_interopRequireDefault(require(\\"@babel/runtime/helpers/possibleConstructorReturn\\"));var _getPrototypeOf2=_interopRequireDefault(require(\\"@babel/runtime/helpers/getPrototypeOf\\"));var _inherits2=_interopRequireDefault(require(\\"@babel/runtime/helpers/inherits\\"));var _react=_interopRequireDefault(require(\\"react\\"));var MyTestComponent=function(_React$Component){(0,_inherits2.default)(MyTestComponent,_React$Component);function MyTestComponent(){(0,_classCallCheck2.default)(this,MyTestComponent);return(0,_possibleConstructorReturn2.default)(this,(0,_getPrototypeOf2.default)(MyTestComponent).apply(this,arguments));}(0,_createClass2.default)(MyTestComponent,[{key:\\"render\\",value:function render(){var opts={test:true,derp:'something',func:function func(){}};return _react.default.createElement(\\"div\\",(0,_extends2.default)({hello:true},opts),_react.default.createElement(_react.default.Fragment,null,_react.default.createElement(\\"span\\",{stuff:true},\\"Hello world!\\"),_react.default.createElement(\\"derp\\",null)));}}]);return MyTestComponent;}(_react.default.Component);"`;

exports[`babel-preset-react-ameliorate should work with plugin options 1`] = `"var _interopRequireDefault=require(\\"@babel/runtime/helpers/interopRequireDefault\\");var _extends2=_interopRequireDefault(require(\\"@babel/runtime/helpers/extends\\"));var _classCallCheck2=_interopRequireDefault(require(\\"@babel/runtime/helpers/classCallCheck\\"));var _createClass2=_interopRequireDefault(require(\\"@babel/runtime/helpers/createClass\\"));var _possibleConstructorReturn2=_interopRequireDefault(require(\\"@babel/runtime/helpers/possibleConstructorReturn\\"));var _getPrototypeOf2=_interopRequireDefault(require(\\"@babel/runtime/helpers/getPrototypeOf\\"));var _inherits2=_interopRequireDefault(require(\\"@babel/runtime/helpers/inherits\\"));var _react=_interopRequireDefault(require(\\"react\\"));var MyTestComponent=function(_React$Component){(0,_inherits2.default)(MyTestComponent,_React$Component);function MyTestComponent(){(0,_classCallCheck2.default)(this,MyTestComponent);return(0,_possibleConstructorReturn2.default)(this,(0,_getPrototypeOf2.default)(MyTestComponent).apply(this,arguments));}(0,_createClass2.default)(MyTestComponent,[{key:\\"render\\",value:function render(){var opts={test:true,derp:'something',func:function func(){}};return __somethingElseBro(this)(\\"div\\",(0,_extends2.default)({hello:true},opts),__somethingElseBro(this)(_react.default.Fragment,null,__somethingElseBro(this)(\\"span\\",{stuff:true},\\"Hello world!\\"),__somethingElseBro(this)(\\"derp\\",null)));}}]);return MyTestComponent;}(_react.default.Component);"`;
